prebid/Prebid.js
# Size Mapping V2

The core feature has been implemented.
- How to test it?

- Edit the *hello_world.html* file, and replace it with an adUnit object containing the sizeConfig V2 specs. You can get hold of those objects from I2I, here. https://github.com/prebid/Prebid.js/issues/4129

- Currently, the workflow (how to reach and activate this module) is not in place. I've currently created a hook on **getBids** function in adapterManager.js, and that hooked function is getting called for client side adapters only, at the present moment.

- I've disabled the function **checkAdUnitSetup** temporarily in **Prebid.js** file to make this work.

- The **hello_world** example has been modified with a simple adUnit setup with sizeMapping V2 specs to check the code. Later, will create a dedicated html example for this new feature.

- Known Bugs
1. Bidder level sizeConfigs are working weirdly. When sizeConfig disables the bid, the bid is coming as undefined. This should not happen.
2. Haven't yet put checks in place to verify if the structure of the sizeConfig object received is correct or not. So, if we give some object in sizeConfig, not properly configured, the behaviour is unstable.

**EDIT**:

- **Q**: How to test it?

- Check the folder **sizeMappingV2** under *integrationExamples -> gpt*, different scenarios has been already defined there. You can edit them to your liking to test edge cases and other scenarios.

- The Workflow is working End-To-End. The feature is complete.

- **checkAdUnitSetup**  in *prebid.js* is hooked onto in **sizeMappingV2** module to check the *sizeConfig* object.

- All *Known Bugs* mentioned above has been resolved.

- Pending Tasks: 
  - Writing unit tests.
  - Documentation in Prebid.org

#### **Known Issue**

If banner and video mediaTypes, both somehow gets filtered out, **adUnits.sizes** is set to an empty array. Due to this, some bidder adapters may complain.
